src/neo-vm/ExecutionEngine.cs

```diff
@@ -42,6 +85,137 @@ public void AddBreakPoint(byte[] script_hash, uint position)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckMaxItemSize(int length) => length <= MaxItemSize;
+
+        /// <summary>
+        /// Check if the is possible to overflow the MaxInvocationStack
+        /// </summary>
+        /// <param name="stack">Stack</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckMaxInvocationStack() => InvocationStack.Count < MaxInvocationStackSize;
+
+        /// <summary>
+        /// Check if the BigInteger is allowed for numeric operations
+        /// </summary>
+        /// <param name="value">Value</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckBigInteger(BigInteger value) => value.ToByteArray().Length <= MaxSizeForBigInteger;
+
+        /// <summary>
+        /// Check if the BigInteger is allowed for numeric operations
+        /// </summary>
+        /// <param name="byteLength">Value</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckBigIntegerByteLength(int byteLength) => byteLength <= MaxSizeForBigInteger;
+
+        /// <summary>
+        /// Check if the number is allowed from SHL and SHR
+        /// </summary>
+        /// <param name="shift">Shift</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckShift(int shift) => shift <= Max_SHL_SHR && shift >= Min_SHL_SHR;
+
+        /// <summary>
+        /// Check if the is possible to overflow the MaxStackSize
+        /// </summary>
+        /// <param name="count">Stack item count</param>
+        /// <param name="strict">Is stack count strict?</param>
+        /// <returns></returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckStackSize(bool strict, int count = 1)
+        {
+            is_stackitem_count_strict &= strict;
+            stackitem_count += count;
+
+            if (stackitem_count < 0) stackitem_count = int.MaxValue;
+            if (stackitem_count <= MaxStackSize) return true;
+            if (is_stackitem_count_strict) return false;
+            stackitem_count = GetItemCount(InvocationStack.SelectMany(p => p.EvaluationStack.Concat(p.AltStack)));
+            if (stackitem_count > MaxStackSize) return false;
+            is_stackitem_count_strict = true;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Decrease stack item count
+        /// </summary>
+        /// <param name="count">Stack item count</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void DecreaseStackItem(int count = 1) => stackitem_count -= count;
```

Comment:
Can we combine `DecreaseStackItem()` and `DecreaseStackItemWithoutStrict()` into `CheckStackSize()`?

For example:

```c#
CheckStackSize(false, -1);
```

Reply:
Because when you decrease never could be greater than the maximum, i thought that be could use a faster method

Reply:
Because when you decrease never could be greater than the maximum, i thought that be could use a faster method

Reply:
```c#
if (count < 0) return true;
```